Engine/Generic/RuleSuppression.cs

```diff
@@ -97,6 +97,15 @@ public string Justification
             set;
         }
 
+        /// <summary>
+        /// Returns the kind of the suppression
+        /// </summary>
+        public SuppressionKind Kind
+        {
+            get;
+            set;
+        }
```

Comment:
What purpose does this field serve? Does PSSA ever set it to anything other than `InSource`?

If the other possible values are for an external tool, like SARIF, then I don't think this property belongs on this object. Instead the external tool should wrap this output and decorate it as it needs. Otherwise we start coupling PSScriptAnalyzer's implementation to unknown external tools.

Reply:
As you know, RuleSuppression includes the necessary info when the user suppresses one violation.

'kind' attribute in RuleSuppression indicates where the suppression is persisted. So I think this property belongs to this object. 

PSSA only supports 'InSource'. According to SARIF-SDK published by MS, SuppressionKind includes: InSource / External / None.

Reply:
> According to SARIF-SDK published by MS, SuppressionKind includes

Yeah, this sounds like it's a SARIF concept rather than a PSSA concept, so should not be in the PSSA code.

Reply:
Hi, Rob, there is another finding:
-ExcludeRule "rulename", is equal to a filter, it can globally suppress all violations related to the rule. Right?
![image](https://user-images.githubusercontent.com/86460533/126431198-bdc108f3-f542-4fb2-802b-2b5ce0693654.png)

so actually, there're two ways to suppress:
1、using [System.Diagnostics.CodeAnalysis.SuppressMessageAttribute('rulename', '')],which is 'InSource'.
2、using -ExcludeRule "rulename",which is 'External'.

if you agree with that, then we need to know where the suppression is persisted: 'InSource'/'External'-->' kind'.

Reply:
Hi, Rob, there is another finding:
![image](https://user-images.githubusercontent.com/86460533/126433655-f315c21e-29a4-44fb-98c4-c10ad3240a24.png)
according to https://github.com/PowerShell/PSScriptAnalyzer/blob/master/README.md
we can use -Settings / -ExcludeRule, which is equal to a filter, it can globally suppress all violations related to the rule. Right?
![image](https://user-images.githubusercontent.com/86460533/126433603-3cfcadee-132a-4a17-9c97-f85d3df35e7b.png)
![image](https://user-images.githubusercontent.com/86460533/126433928-9a1a03d0-78c4-4748-a97a-caad756a12fc.png)


so actually, there're two ways to suppress:
1、using [System.Diagnostics.CodeAnalysis.SuppressMessageAttribute('rulename', '')],which is 'InSource'.
2、using -Settings/-ExcludeRule , which is 'External'.

if you agree with that, then we need to know where the suppression is persisted: 'InSource'/'External'-->' kind'.

Reply:
Well `-ExcludeRule` isn't a form of suppression to PSSA, it's just a reconfiguration of what rules are run. Conceptually, PSSA takes a series of rules, runs them on scripts and emits diagnostics, unless those diagnostics are suppressed. But excluded rules are never run and never emit any diagnostics, so no diagnostics from them are suppressed, so that's not considered suppression.

If you use `-ExcludeRule` today, you'll see that diagnostics that you would have gotten aren't recorded anywhere as suppressed.

Reply:
Get it. I'll remove `kind` out of RuleSuppression. Based on PSSA only support `InSource` suppress, ConvertToSARIF can directly add `InSource` to its suppressedRecord suppressions-kind.